spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/test/java/com/example/SpannerRepositoryTests.java

```diff
@@ -0,0 +1,60 @@
+package com.example;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assume.assumeThat;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.cloud.gcp.data.spanner.core.admin.SpannerDatabaseAdminTemplate;
+import org.springframework.cloud.gcp.data.spanner.core.admin.SpannerSchemaUtils;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringRunner;
+
+@RunWith(SpringRunner.class)
+@TestPropertySource("classpath:application-test.properties")
+@SpringBootTest(classes = {SpannerRepositoryExample.class})
+@ActiveProfiles(profiles = "repository-example")
+public class SpannerRepositoryTests {
+
+	@Autowired
+	private TraderRepository traderRepository;
+
+	@Autowired
+	private TradeRepository tradeRepository;
+
+	@Autowired
+	private SpannerSchemaUtils spannerSchemaUtils;
+
+	@Autowired
+	private SpannerDatabaseAdminTemplate spannerDatabaseAdminTemplate;
+
+	@Autowired
+	private CommandLineRunner commandLineRunner;
+
+	@BeforeClass
+	public static void checkToRun() {
+		assumeThat(
+				"Spanner integration tests are disabled. "
+						+ "Please use '-Dit.spanner=true' to enable them. ",
+				System.getProperty("it.spanner"), is("true"));
+	}
+
+	@After
+	public void cleanupSpannerTables() {
+		this.tradeRepository.deleteAll();
+		this.traderRepository.deleteAll();
+	}
+
+	@Test
+	public void verifySpannerApplicationLoadsRows() {
+		// Verifies that there are 3 items in the Trader Repository table,
+		assertThat(traderRepository.count()).isEqualTo(3);
```

Comment:
What about testing `Trades`?
Also, you probably want to test `http://localhost:8080/trades` and see if it returns what is expected.
Essentially, the test should validate as much as possible what the user would do and see in the description of [steps](https://github.com/spring-cloud/spring-cloud-gcp/tree/master/spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample).

Reply:
@meltsufin  Regarding "What about testing Trades". We aren't testing if Spring Data Spanner works; that is heavily tested in its own module. We're just testing if the sample ran, and I think this accomplishes that.

Reply:
I think redundancy in the tests is OK. Given that we already start up the application, it's not all that much more execution cost to add these tests. It will also ensure that we don't regress the sample code itself, like enabling Spring Data REST.

Reply:
Done. I improved tests to verify the actions done in each demo.